extmod/vfs_fat.c

```diff
@@ -250,6 +250,37 @@ STATIC mp_obj_t fat_vfs_stat(mp_obj_t vfs_in, mp_obj_t path_in) {
 }
 STATIC MP_DEFINE_CONST_FUN_OBJ_2(fat_vfs_stat_obj, fat_vfs_stat);
 
+/// \function statvfs(path)
+/// Get the status of a VFS.
+STATIC mp_obj_t fat_vfs_statvfs(mp_obj_t vfs_in, mp_obj_t path_in) {
+    (void)vfs_in;
+    const char *path = mp_obj_str_get_str(path_in);
+
+    FATFS *fatfs;
+    DWORD nclst;
+    FRESULT res = f_getfree(path, &nclst, &fatfs);
+    if (FR_OK != res) {
+        nlr_raise(mp_obj_new_exception_arg1(&mp_type_OSError,
+          MP_OBJ_NEW_SMALL_INT(fresult_to_errno_table[res])));
+    }
+
+    mp_obj_tuple_t *t = MP_OBJ_TO_PTR(mp_obj_new_tuple(10, NULL));
+
+    t->items[0] = MP_OBJ_NEW_SMALL_INT(fatfs->csize * 512); // f_bsize - block size
```

Comment:
I think the 512 should be fatfs->ssize.  On stmhal ssize does not exist and 512 should be used instead, but on esp8266 you need to use ssize because it has a larger block size (4096 bytes).